crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -60,11 +64,22 @@ pub(crate) fn do_finalize_domain_current_epoch<T: Config>(
     })
 }
 
+/// Result holding after `operator_take_reward_tax_and_stake`
+pub(crate) struct OperatorTakeRewardTaxResult {
```

Comment:
Usually in Rust, structs named Result have an Ok and Error case. A better name here would be:
```suggestion
pub(crate) struct OperatorTakeRewardTaxOutcome {
```